src/python/pants/python/python_setup.py

```diff
@@ -241,7 +243,7 @@ def expand_interpreter_search_paths(cls, interpreter_search_paths, pyenv_root_fu
                     from_pexrc = special_paths
                 expanded.extend(special_paths)
             else:
-                expanded.append(s)
+                expanded.append(os.path.expandvars(s))
```

Comment:
Please add a TODO here:
```suggestion
                # TODO: Use the API provided by #9615 to fill in specific supported variables here.
                expanded.append(os.path.expandvars(s))
```